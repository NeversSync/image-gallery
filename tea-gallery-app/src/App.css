body {
  margin: 0 auto;
  padding: 0;
  font-family: sans-serif;
}

li {
  list-style-type: none;
}

.wrapper {
  text-align: center;
}

.app-header {
  text-align: center;
}

/*Thumbnail styles*/

.thumbnail-list { /*why is there a mysterious padding-left?*/
  max-width: 90%;
  margin: 0 auto;
}

.thumbnail-wrapper {
  display: inline-block;
  width: 30%;
}

.thumbnail-li {
  padding: 0 auto;
}

.thumbnail {
  width: 90%;
  margin: 8% auto;
}

/*List styles*/

.list-item-wrapper {
  border: 1px solid grey;
  border-radius: 10px;
  padding: 1%;
  width: 30%;
  margin: 2% auto;
  text-align: left;
}

.image-list-item {
  margin: .75%;
}

/*Gallery styles*/

.gallery-img-wrapper {
  width: 60%;
  margin: 0 auto;
}

.gallery-img {
  width: 100%;
}

.next-button {
  float: right;
}

.back-button {
  float: left;
}