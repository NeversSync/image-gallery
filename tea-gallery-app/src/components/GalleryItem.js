import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DeleteButton from './DeleteButton';

class GalleryItem extends Component {

  static PropTypes = {
    images: PropTypes.array.required,
    onDeleteImage: PropTypes.func.required
  };

  constructor(props) {
    super(props);

    this.state = {
      index: 0
    };
  }

  changeImage(increment) {
    const images = this.props.images;
    let index = this.state.index;
    const image = images[index];

    index += increment;
    if (index === images.length) index = 0;
    if (index === -1 || isNaN(index)) index = images.length - 1;

    this.setState({ index: index });
    this.setState({ images: image });
  }

  render() {
    const { images} = this.props;
    return (
      <div className="gallery-img-wrapper">
        <img className="gallery-img" src={images[this.state.index].url} alt="../img/broken-glass.jpg"></img>
        <button className="back-button" onClick={() => this.changeImage(-1)}>Back</button>
        <button className="next-button" onClick={() => this.changeImage(+1)}>Next</button> {/* add arrow img?*/}
        <DeleteButton
          images={images}
          onDeleteImage={() => this.props.onDeleteImage(images[this.state.index]._id)}
        />
      </div>
    );
  }
}

export default GalleryItem;