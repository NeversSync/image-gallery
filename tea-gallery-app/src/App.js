import React, { Component } from 'react';
import './App.css';
import Wrapper from './components/Wrapper';
import imageApi from './data.js';

class App extends Component {
  constructor() {
    super();

    this.state = {
      images: null,
    };

    this.onDeleteImage = this.onDeleteImage.bind(this);
  }
  componentDidMount() {
    imageApi.get()
      .then(images => {
        this.setState({
          images
        });
      })
      .catch(error => {
        this.setState({
          error
        });
      });
  }

  onDeleteImage(id) {
    imageApi.deleteImage(id)
      .then(() => {
        const images = this.state.images;
        const index = images.findIndex(img => img._id === id);
        images.splice(index, 1);
        this.setState({ images });
      });
  }

  render() {
    return (
      <div className="App" >
        <div className="app-header">
          <h2>Welcome to the Cha'llery</h2>
          <h3>A photo gallery of <strong>èŒ¶</strong></h3>
        </div>
        {this.state.images &&
          <Wrapper
            images={this.state.images}
            onDeleteImage={this.onDeleteImage}
          />
        }
      </div >
    );
  }
}

export default App;
