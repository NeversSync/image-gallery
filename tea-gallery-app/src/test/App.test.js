import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('App', () => {
  it('renders the app', () => {
    const component = renderer.create(<App/>);
import GalleryItem from '../components/GalleryItem';

const images = [
  {
    title: 'tea leaves in teapot',
    description: 'what nice leaves',
    url: '/../public/img/teainpot.jpg'
  },
  {
    title: 'Tetsubin over charcoal',
    description: 'what a nice tetsubin',
    url: '/../public/img/tetsubin-charcoal.jpg'
  },
  {
    title: 'Teapot display',
    description: 'what a nice teapot display',
    url: '/../public/img/teapot-display.jpg'
  }
];

describe('Gallery Images', () => {
  it('pulls up gallery view', () => {
    const component = renderer.create(<GalleryItem images={images} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});