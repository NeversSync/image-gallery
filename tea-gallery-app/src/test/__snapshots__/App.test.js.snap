// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app 1`] = `
<div
  className="App"
>
  <div
    className="app-header"
  >
    <h2>
      Welcome to the Cha'llery
    </h2>
    <h3>
      A photo gallery of 
      <strong>
        èŒ¶
      </strong>
    </h3>
  </div>
  <form
    className="add-image"
    onSubmit={[Function]}
  >
    <label>
      Image name
      <input
        name="title"
        onChange={[Function]}
      />
    </label>
    <label>
      Description
      <input
        name="description"
        onChange={[Function]}
      />
    </label>
    <label>
      URL
      <input
        name="url"
        onChange={[Function]}
      />
    </label>
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
